src/components/MagicCodeInput.js

```diff
@@ -149,7 +149,7 @@ function MagicCodeInput(props) {
 
         // We have not added the editIndex as the dependency because we don't want to run this logic after focusing on an input to edit it after the user has completed the code.
```

Comment:
@hoangzinh Let's add a comment like this so that we don't end up adding this again.